### PR TITLE
[ADMIN] Add ElectedRepresentative zones

### PR DESCRIPTION
**Pull request by illusionOfParadise**
After deploying this PR we should import EPCI data to `epci` using data from `data.en-marche.fr` project. 
And run this script:

```SQL
INSERT INTO enmarche.elected_representative_zone_category (id, name) VALUES 
(1, 'Ville'),
(2, 'EPCI'),
(3, 'Département'),
(4, 'Région'),
(5, 'Circonscription'),
(6, 'Corse'),
(7, 'FDE')
;

INSERT INTO elected_representative_zone (name, category_id) SELECT CONCAT(name,' (',postal_codes,')'), 1 FROM cities;
INSERT INTO elected_representative_zone (name, category_id) SELECT name, 2 FROM epci WHERE name != '';
INSERT INTO elected_representative_zone (name, category_id) SELECT CONCAT(name,' (',code,')'), 3 FROM department;
INSERT INTO elected_representative_zone (name, category_id) SELECT name, 4 FROM region;
INSERT INTO elected_representative_zone (name, category_id) SELECT CONCAT(name,', ',number, IF(number = 1,'ère', 'ème'), ' circonscription (', LEFT(code, LENGTH(code) -3),'-',RIGHT(code, 2), ')'), 5 FROM districts;
INSERT INTO elected_representative_zone (name, category_id) VALUES ('Corse', 6);
INSERT INTO elected_representative_zone (name, category_id) VALUES ('Français de l\'Étranger', 7);
```

### CODE REVIEW

File: src/Repository/ElectedRepresentative/ZoneRepository.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                $qb->setParameter('category_'.$key, $category);
+            }
+
+            $qb->andWhere($zoneExpression);
+
+            return $qb->getQuery()->getResult();
+        } else {
+            return [];
+        }
+    }
```
Comment (ottaviano) [inline src/Repository/ElectedRepresentative/ZoneRepository.php]:
```suggestion
    public function findForAutocomplete(string $zone, string $type, int $limit = 10): array
    {
        $mandateType = array_search($type, MandateTypeEnum::toArray());
        $categories = ZoneCategory::ZONES[$mandateType];

        if (!\count($categories)) {
            return [];
        }

        return $this->createQueryBuilder('zone')
            ->innerJoin('zone.category', 'category')
            ->where('zone.name LIKE :name')
            ->andWhere('category.name IN (:categories)')
            ->setParameters([
                'name' => $zone.'%',
                'categories' => $categories,
            ])
            ->setMaxResults($limit)
            ->getQuery()
            ->getResult()
        ;
    }
```

File: templates/form_theme_admin.html.twig [hunk 2]
```diff
@@ -167,6 +167,36 @@
             $('div.mandates-collection .sonata-collection-add').on('sonata-collection-item-added', function () {
                 $("select[id$='_isElected'], div.icheckbox_square-blue input[id$='_onGoing']").trigger('ifToggled change');
             });
+
+            $("input[id$='_zone']").each(function (i, el) {
+                el = $(el);
+                el.autocomplete({
+                    source: function(request, response) {
+                        var elementId = el.attr('id');
+                        console.log(elementId);
```
Comment (ottaviano) [inline templates/form_theme_admin.html.twig]: 🙈

File: templates/form_theme_admin.html.twig [hunk 3]
```diff
@@ -166,7 +166,49 @@
...
             });
+
+            var addAutocomplete = function () {
+                $("input[id$='_zone_autocomplete']").each(function (i, el) {
+                    el = $(el);
+                    el.autocomplete({
+                        source: function(request, response) {
+                            var elementId = el.attr('id');
+                            var id = elementId.substr(0, elementId.indexOf('_zone_autocomplete'));
+                            console.log(id);
```
Comment (ottaviano) [inline templates/form_theme_admin.html.twig]: 👀

**Comment by illusionOfParadise** [comment]
[ch11028]

**Pull request merged**